src/compilation/compiler.rs

```diff
@@ -203,13 +242,16 @@ impl Compiler {
             LLVMSetValueName2(param_value_ref, param_name.as_ptr(), param_name_len);
         }
 
-        // TODO: Remove variable shadowing
+        // TODO: Design: Remove variable shadowing
```

Comment:
I actually have no clue what this is referring to...

But I need to refresh my mind on how we handle scopes. Because we have a seperate vars array and a funcs array in scope_manager. This means that you could theoretically define a variable with the same name as the function, which I am not sure we want to allow.

Reply:
Right, yes. Ok let's look at this when we go through all our TODOs.